src/App.tsx

```diff
@@ -1,27 +1,67 @@
-import React from 'react';
+import React, { useState } from 'react';
 import './App.scss';
 import { GoodsList } from './GoodsList';
 
-// import { getAll, get5First, getRed } from './api/goods';
-// or
+import { getAll, get5First, getRedGoods } from './api/goods';
+
 // import * as goodsAPI from './api/goods';
+import { Good } from './types/Good';
+
+export const App: React.FC = () => {
+  const [allGoods, setAllGoods] = useState<Good[]>([]);
+
+  const handleAllGoods = async (): Promise<void> => {
+    try {
+      const goods = await getAll();
+
+      setAllGoods(goods);
+    } catch (error) {
+      // eslint-disable-next-line no-console
+      console.error('No gooods', error);
+    }
+  };
+
+  const handle5Goods = async () => {
+    const goods = await get5First();
+
+    setAllGoods(goods);
+  };
+
+  const handleRedGoods = async () => {
+    const goods = await getRedGoods();
+
+    setAllGoods(goods);
+  };
```

Comment:
You repeat the same actions here. You can create a separate method and pass the callback there

Reply:
it'll looks like:

```
const handleFetchGoods = async (callback) => {
     try {
      const fetchedGoods = await callback();
      
      setGoods(fetchedGoods);
    } catch (error) {
      setError(error.message || 'No goods');
    }
}
```

and you will use this helper like this:
```
onClick={() => handleFetchGoods(getAll)}
onClick={() => handleFetchGoods(get5First)}
onClick={() => handleFetchGoods(getRedGoods)}
```